src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -640,6 +642,470 @@ public static SyntaxToken XmlTextLiteral(SyntaxTriviaList leading, string text,
+        /// <param name="name">The name of the xml text attribute.</param>
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(string name, params SyntaxToken[] textTokens)
+        {
+            return XmlTextAttribute(XmlName(name), SyntaxKind.DoubleQuoteToken, TokenList(textTokens));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml text attribute.
+        /// </summary>
+        /// <param name="name">The name of the xml text attribute.</param>
+        /// <param name="quoteKind">The kind of the quote token to be used to quote the value (e.g. " or ').</param>
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(string name, SyntaxKind quoteKind, SyntaxTokenList textTokens)
+        {
+            return XmlTextAttribute(XmlName(name), SyntaxKind.DoubleQuoteToken, textTokens);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml text attribute.
+        /// </summary>
+        /// <param name="name">The name of the xml text attribute.</param>
+        /// <param name="quoteKind">The kind of the quote token to be used to quote the value (e.g. " or ').</param>
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(XmlNameSyntax name, SyntaxKind quoteKind, SyntaxTokenList textTokens)
+        {
+            return XmlTextAttribute(name, Token(quoteKind), textTokens, Token(quoteKind))
+                .WithLeadingTrivia(Whitespace(" "));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="localName">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(string localName, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlMultiLineElement(XmlName(localName), content);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="name">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(XmlNameSyntax name, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlElement(
+                XmlElementStartTag(name),
+                content.Insert(0, XmlNewLine("\r\n")).Add(XmlNewLine("\r\n")),
```

Comment:
@mattwar @AnthonyDGreen Now, this code concerns me a bit. The XmlMultiLineElement factory method is the first method that generates multiline code (as far as I can see). And I need to use this method within the higher level factory methods like XmlSummaryElement. So I have to give some kind of new line text as parameter value to the XmlNewLine factory method. So I decided to use "\r\n". Alternatively, we have to pass a text parameter for the new line text to the XmlSummaryElement factory method as well. What would you suggest to handle this situation?